### ISSUE TITLE
Detect blank lines between YARD documentation and method definition

### ISSUE THREAD

**Issue by nijikon** [mensfeld/yard-lint#30]
## Problem

Blank lines between YARD documentation and method definitions cause two distinct issues:

1. **Single blank line** — Documentation is still associated, but creates unnecessary whitespace (convention issue)
2. **Two or more blank lines** — Documentation is completely detached and lost (functional issue)

Currently, yard-lint does not detect either case.

## Verified YARD Behavior

| Blank lines between doc and def | Documentation associated? | Issue type |
|--------------------------------|---------------------------|------------|
| 0 | Yes | None |
| 1 | Yes | Convention |
| 2+ | **No** (documentation lost) | Error |

### Single blank line (convention issue, not detected)

```ruby
# @param organization_id [String]
# @param id [String]
#
# @return [Pet]

def call(organization_id, id)
  # ...
end
```

Documentation works, but the blank line is unnecessary and inconsistent.

### Multiple blank lines (error, not detected)

```ruby
# @param organization_id [String]
# @param id [String]
#
# @return [Pet]


def call(organization_id, id)
  # ...
end
```

Documentation is lost — method returns `Object` instead of `Pet` and appears as undocumented in YARD stats.

### Correct (no blank line)

```ruby
# @param organization_id [String]
# @param id [String]
#
# @return [Pet]
def call(organization_id, id)
  # ...
end
```

## Proposed Solution

Add a new error type: `BlankLineBeforeDefinition`

With two severity levels:

| Condition | Severity | Message |
|-----------|----------|---------|
| 1 blank line | Convention/Warning | "Unnecessary blank line between documentation and definition" |
| 2+ blank lines | Error | "Documentation detached from definition (2+ blank lines causes YARD to ignore documentation)" |

## Rationale

1. **Silent documentation loss** — Multiple blank lines cause documentation to be silently ignored by YARD
2. **Consistency** — Single blank lines create inconsistent formatting across the codebase
3. **Common mistake** — Developers may add blank lines for visual separation without realizing the implications
4. **Hard to detect manually** — The code looks correctly documented at a glance, making this issue easy to miss in code review
5. **Complements existing checks** — This pairs well with the existing undocumented code detection by catching a root cause

## Configuration

```yaml
BlankLineBeforeDefinition:
  enabled: true
  # Severity for single blank line (convention issue)
  single_blank_line_severity: warning  # or: error, info, ignore
  # Severity for multiple blank lines (documentation loss)
  multiple_blank_lines_severity: error
```

## Edge Cases

- Should detect before `def`, `class`, `module`, and constant assignments
- Should not false-positive on intentionally orphaned comments (though distinguishing intent may be difficult)
